### PR TITLE
Bouton rewrite

### PR DESCRIPTION
**Pull request by fraincs**
Issue:
- Button CSS was repetitive.
- Outline were not working on any background other than white

## What I did
- Removed repetitive CSS declarations
- Used another technique to display an outline on focus.

### CODE REVIEW

File: packages/react-components/src/button/src/Button.css [hunk 1]
```diff
@@ -18,12 +18,14 @@
     cursor: pointer;
     outline: none;
     font-weight: 500;
+    justify-content: center;
```
Comment (patricklafrance) [inline packages/react-components/src/button/src/Button.css:21]: Doesn't make sense to me to set `justify-content` when the element display is not `flex` or `grid` ?

File: packages/react-components/src/index.css [hunk 2]
```diff
@@ -1,16 +1,13 @@
-:root {
-    --o-ui-disabled-opacity: 0.4;
-    --o-ui-pill-border-radius: 999px;
-    --o-ui-rounded-border-radius: var(--scale-alpha);
-    --o-ui-circular-border-radius: 100%;
-}
-
 [class^="o-ui-"] {
     --o-ui-focus-ring-inset-sm: 0 0 0 1px var(--o-ui-focus-ring-color);
```
Comment (patricklafrance) [inline packages/react-components/src/index.css:2]: Are these var still used since you updated how the focus ring is done?

File: packages/react-components/src/checkbox/src/Checkbox.jsx [hunk 3]
```diff
@@ -152,7 +151,7 @@ export function InnerCheckbox(props) {
             {...rest}
             {...wrapperProps}
         >
-            <VisuallyHidden {...inputProps} />
+            <input {...inputProps} />
```
Comment (patricklafrance) [inline packages/react-components/src/checkbox/src/Checkbox.jsx:154]: Why? I would really prefer if you don't do this.

File: packages/react-components/src/button/src/Button.css [hunk 4]
```diff
@@ -33,25 +35,71 @@
...
+}
+
+/* STATES | FOCUS | FOCUS RING */
+.o-ui-button:after {
+    opacity: 0;
+    content: "";
+    top: -2px;
+    left: -2px;
+    right: -2px;
+    bottom: -2px;
```
Comment (patricklafrance) [inline packages/react-components/src/button/src/Button.css:52]: How come our focus ring is now that complex? Used to be a simple box-shadow.

These focus ring position values should be extracted to vars in `index.css`

File: packages/react-components/src/button/src/Button.css [hunk 5]
```diff
@@ -429,11 +364,14 @@ a.o-ui-button {
...
-.o-ui-button-ghost:focus,
-.o-ui-button-ghost.o-ui-button-focus {
-    box-shadow: var(--o-ui-focus-ring-inset-md);
-    z-index: 2;
-    outline: none;
+.o-ui-button-ghost:focus:after,
+.o-ui-button-ghost.o-ui-button-focus:after {
+    opacity: 1;
+    /* inner */
+    top: 1px;
```
Comment (patricklafrance) [inline packages/react-components/src/button/src/Button.css:371]: These position values should also be extracted to `index.css`

File: packages/react-components/src/checkbox/src/Checkbox.jsx (hunk 3 above)
Comment (patricklafrance) [reply packages/react-components/src/checkbox/src/Checkbox.jsx:154]: React should throw an exception stating it doesn't know to render the prop `as`.  I am not sure why I don't see this exception in SB preview

File: packages/react-components/src/button/src/Button.css [hunk 6]
```diff
@@ -33,25 +35,71 @@
...
+    content: "";
+    top: -2px;
+    left: -2px;
+    right: -2px;
+    bottom: -2px;
+    display: block;
+    position: absolute;
+    box-shadow: 0 0 0 2px var(--o-ui-focus-ring-color);
+    border-radius: var(--o-ui-pill-border-radius);
+    transition: opacity .2s ease-in;
```
Comment (patricklafrance) [inline packages/react-components/src/button/src/Button.css:57]: Should also be a CSS var?

File: packages/react-components/src/checkbox/src/Checkbox.css [hunk 7]
```diff
@@ -240,13 +261,24 @@
...
 /* SIZING | LARGE */
 /* SIZING | LARGE | LABEL */
 .o-ui-checkbox-lg .o-ui-checkbox-box:before {
     width: var(--scale-foxtrot);
     height: var(--scale-foxtrot);
 }
 
+.o-ui-checkbox-lg input {
+    width: 32px;
+    height: 32px;
```
Comment (patricklafrance) [inline packages/react-components/src/checkbox/src/Checkbox.css:279]: Should use `--scale-golf`

File: packages/react-components/src/button/src/Button.css (hunk 1 above)
Comment (fraincs) [reply packages/react-components/src/button/src/Button.css:21]: This was meant to be as generic as possible, I agree that I'd rather put it on `.o-ui-button:not(.o-ui-icon-button)` which is where we set buttons to a display of `inline-grid`

File: packages/react-components/src/index.css (hunk 2 above)
Comment (fraincs) [reply packages/react-components/src/index.css:2]: `--o-ui-rounded-border-radius` is still used for inner focus-rings (the ones that are not offset)

File: packages/react-components/src/checkbox/src/Checkbox.jsx (hunk 3 above)
Comment (fraincs) [reply packages/react-components/src/checkbox/src/Checkbox.jsx:154]: > Why? I would really prefer if you don't do this and keep VisuallyHidden.
The only way to have an offset on a checkbox since both pseudos are already in use for the checkmark and the border.

File: packages/react-components/src/button/src/Button.css (hunk 4 above)
Comment (fraincs) [reply packages/react-components/src/button/src/Button.css:52]: Since box-shadow does not supports an offset, on any background other than white you would see a 2px solid white border.

File: packages/react-components/src/checkbox/src/Checkbox.css (hunk 7 above)
Comment (fraincs) [reply packages/react-components/src/checkbox/src/Checkbox.css:279]: the logic is 2x the border width (2 sides to a box) + 2 times the offset

File: packages/react-components/src/checkbox/src/Checkbox.css [hunk 8]
```diff
@@ -1,4 +1,6 @@
 .o-ui-checkbox {
+    --o-ui-focus-ring-stroke-width: 2px;
+    --o-ui-indeterminate-stroke-width-md: 5px;
```
Comment (patricklafrance) [inline packages/react-components/src/checkbox/src/Checkbox.css:3]: Why `-md` suffix for the for the indeterminate but not for `--o-ui-focus-ring-stroke-width`. Don't care if we add it or remove it but if it should be consistant if there isn't any good reason not to be.

File: packages/react-components/src/checkbox/src/Checkbox.css [hunk 9]
```diff
@@ -7,7 +9,7 @@
     line-height: 1;
     max-width: 100%;
     width: max-content;
-    height: max-content;
```
Comment (patricklafrance) [inline packages/react-components/src/checkbox/src/Checkbox.css:10]: Why did you remove the max-content? Ideally we would keep this for every components

File: packages/react-components/src/checkbox/stories/Checkbox.stories.mdx [hunk 10]
```diff
@@ -84,7 +84,7 @@ A checkbox can be indeterminate.
 
 <Preview>
     <Story name="indeterminate">
-        <Checkbox defaultIndeterminate>Milky Way</Checkbox>
+        <Checkbox size="lg" defaultIndeterminate>Milky Way</Checkbox>
```
Comment (patricklafrance) [inline packages/react-components/src/checkbox/stories/Checkbox.stories.mdx:87]: Derp here with `size="lg"`

File: packages/react-components/src/index.css [hunk 11]
```diff
@@ -1,16 +1,16 @@
...
     --o-ui-focus-ring-inset-md: 0 0 0 2px var(--o-ui-focus-ring-color);
     --o-ui-focus-ring-inset-lg: 0 0 0 4px var(--o-ui-focus-ring-color);
     --o-ui-focus-ring: 0 0 0 2px var(--white), 0 0 0 4px var(--o-ui-focus-ring-color);
+    --o-ui-focus-ring-offset: 2px;
+    --o-ui-focus-ring-inner-offset: 1px;
```
Comment (patricklafrance) [inline packages/react-components/src/index.css:7]: `inner-offset` doesn't make much sense to me.

File: packages/react-components/src/index.css [hunk 12]
```diff
@@ -1,16 +1,16 @@
...
-    --o-ui-disabled-opacity: 0.4;
-    --o-ui-pill-border-radius: 999px;
-    --o-ui-rounded-border-radius: var(--scale-alpha);
-    --o-ui-circular-border-radius: 100%;
-}
-
 [class^="o-ui-"] {
     --o-ui-focus-ring-inset-sm: 0 0 0 1px var(--o-ui-focus-ring-color);
     --o-ui-focus-ring-inset-md: 0 0 0 2px var(--o-ui-focus-ring-color);
     --o-ui-focus-ring-inset-lg: 0 0 0 4px var(--o-ui-focus-ring-color);
```
Comment (patricklafrance) [inline packages/react-components/src/index.css:4]: We do have an `outset` and an `inset` focus-ring. Therefore I would expect to have var names that contains `inset` & `ouset`. We do have vars for `inset` but I don't see any for `outset`.

File: packages/react-components/src/input/src/Input.css [hunk 13]
```diff
@@ -188,7 +187,7 @@
 }
 
 /* BUTTON */
-.o-ui-input-button {
+.o-ui-input-has-button .o-ui-input-button {
```
Comment (patricklafrance) [inline packages/react-components/src/input/src/Input.css:190]: Not sure why would need to add `has-button` here? The class `o-ui-input-button` would only be available there IS a button. There IMO it's redundant and useless.

File: packages/react-components/src/index.css (hunk 11 above)
Comment (fraincs) [reply packages/react-components/src/index.css:7]: this is opposed to the outer offset, how would you name this variable since it's use is only to "insert" the focus ring in the element.

File: packages/react-components/src/index.css (hunk 12 above)
Comment (fraincs) [reply packages/react-components/src/index.css:4]: That is because the outset focus ring is not using a box shadow

Comment (patricklafrance) [reply packages/react-components/src/index.css:4]: Well as far as I understand all this we do have a var like `--o-ui-focus-ring-offset: 2px;` that exist to be used as part of an `outset` focus-ring?

`--o-ui-focus-ring-outset-offset: 2px`

File: packages/react-components/src/index.css (hunk 11 above)
Comment (patricklafrance) [reply packages/react-components/src/index.css:7]: Make sense, but how was I supposed to understand this by looking at these var names?

`--o-ui-focus-ring-offset: 2px;` should somehow include outer then

File: packages/react-components/src/index.css (hunk 12 above)
Comment (patricklafrance) [reply packages/react-components/src/index.css:4]: After reviewing other comments it should be something like:

```
--o-ui-focus-ring-outset-outer-offset: 2px;
--o-ui-focus-ring-outset-inner-offset: 1px;
```

File: packages/react-components/src/index.css (hunk 11 above)
Comment (fraincs) [reply packages/react-components/src/index.css:7]: totall;
